allennlp/data/fields/array_field.py

```diff
@@ -1,68 +1,78 @@
-from typing import Dict
+from typing import Dict, Any, Union, Optional
 
-import numpy
 import torch
+import numpy as np
 from overrides import overrides
 
 from allennlp.data.fields.field import Field
 
 
-class ArrayField(Field[numpy.ndarray]):
+class ArrayField(Field[torch.Tensor]):
```

Comment:
Given the changes you made, I'd be tempted to rename this and add something like `ArrayField = TensorField` for backwards compatibility.